Panic when creating snapshot

Submitting a snapshot with :

![image](https://github.com/rancher/opni/assets/56211853/9d7d91f1-ef89-4765-ac5b-eb74f549077b)

I'm suspecting additional index is required

This is the request payload

UI payload : `snapshot10*/5**`

Gives the following panic:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xabd078]

goroutine 993 [running]:
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.func1()
	go.opentelemetry.io/otel/sdk@v1.16.0/trace/span.go:383 +0x25
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(0xc001074000, {0x0, 0x0, 0x42a1d00?})
	go.opentelemetry.io/otel/sdk@v1.16.0/trace/span.go:415 +0x975
panic({0x23ff520?, 0x4256c50?})
	runtime/panic.go:914 +0x21f
log/slog.(*Logger).Handler(...)
	log/slog/logger.go:89
log/slog.(*Logger).Enabled(0x0?, {0x2c68818?, 0x42d3e40?}, 0x10?)
	log/slog/logger.go:137 +0x18
log/slog.(*Logger).log(0x0, {0x2c68818, 0x42d3e40}, 0x8, {0x28b7dfb, 0x24}, {0x0, 0x0, 0x0})

```

### Fix 

restart gateway

--- a/pkg/opensearch/reconciler/reconciler.go
+++ b/pkg/opensearch/reconciler/reconciler.go
@@ -1281,6 +1281,9 @@ func (r *Reconciler) snapshotPolicyNeedsUpdate(new, current types.SnapshotManage
 		return false
 	}
 
+	if current.Deletion == nil {
+		return false
+	}
 	lg.Info("new deletion is empty, but current isn't")
 	if !reflect.DeepEqual(new.Creation.Schedule, current.Deletion.Schedule) {
 		lg.Info(fmt.Sprintf("deletion schedule is different; current %+v new %+v", current.Deletion.Schedule, new.Creation.Schedule))

--- a/plugins/logging/pkg/errors/errors.go
+++ b/plugins/logging/pkg/errors/errors.go
@@ -24,6 +24,7 @@ var (
 	ErrReplicasZero            = utilerrors.New(codes.InvalidArgument, errors.New("replicas must be a positive nonzero integer"))
 	ErrAlreadyExists           = utilerrors.New(codes.AlreadyExists, errors.New("object with that name already exists"))
 	ErrObjectNotFound          = utilerrors.New(codes.NotFound, errors.New("object not found"))
+	ErrSnapshotName            = utilerrors.New(codes.InvalidArgument, errors.New("snapshot name is required"))
 )
 
 func WrappedGetPrereqFailed(inner error) *utilerrors.GRPCError {

--- a/plugins/logging/pkg/gateway/drivers/management/kubernetes_manager/kubernetes_manager.go
+++ b/plugins/logging/pkg/gateway/drivers/management/kubernetes_manager/kubernetes_manager.go
@@ -6,6 +6,8 @@ import (
 	"os"
 	"time"
 
+	"log/slog"
+
 	"github.com/lestrrat-go/backoff/v2"
 	"github.com/rancher/opni/apis"
 	opnicorev1beta1 "github.com/rancher/opni/apis/core/v1beta1"
@@ -28,7 +30,6 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/util/retry"
-	"log/slog"
 	opsterv1 "opensearch.opster.io/api/v1"
 	"opensearch.opster.io/pkg/helpers"
 	ctrl "sigs.k8s.io/controller-runtime"
@@ -53,7 +54,7 @@ type KubernetesManagerDriver struct {
 type KubernetesManagerDriverOptions struct {
 	OpensearchCluster *opnimeta.OpensearchClusterRef `option:"opensearchCluster"`
 	K8sClient         client.Client                  `option:"k8sClient"`
-	Logger            *slog.Logger
+	Logger            *slog.Logger                   `option:"logger"`
 }
 
 func NewKubernetesManagerDriver(options KubernetesManagerDriverOptions) (*KubernetesManagerDriver, error) {
@@ -433,6 +434,10 @@ func (d *KubernetesManagerDriver) CreateOrUpdateSnapshotSchedule(
 		}
 		d.Logger.Error(fmt.Sprintf("failed to list opensearch repositories: %v", err))
 		return k8sutilerrors.GRPCFromK8s(err)
+	}
+
+	if snapshot.Ref == nil {
+		return loggingerrors.ErrSnapshotName
 	}
 
 	return d.createOrUpdateRecurringSnapshot(ctx, snapshot, defaultIndices, repo)

